src/autodev.ts

```diff
@@ -75,44 +95,41 @@ const autoDev = async (): Promise<void> => {
   await exec('git fetch')
   await exec(`git config --global user.email "${email}"`)
   await exec(`git config --global user.name "${user}"`)
-  await exec(`git checkout ${branch}`)
-  await exec(`git reset --hard origin/${base}`)
+  const commitDate = await execAndSlurp(
+    `git show -s --format='%ci' origin/${base}`
+  )
+
+  await exec(`git checkout ${base}`)
 
   if (pulls.length === 0) {
-    if (await hasDiff('HEAD', `origin/${branch}`)) {
-      await exec('git push -f')
-      info(
-        `🎉 No Pull Requests found. Pushed changes, because "${branch}" and "${base}" diverged.`
-      )
-    } else {
-      info('🎉 No Pull Requests found. Nothing to merge.')
-    }
-    return
+    info('🎉 No Pull Requests found. Nothing to merge.')
+  } else {
+    const message = await merge(
+      base,
+      pulls,
+      updateComment,
+      updateLabel,
+      commitDate
+    )
+    info(message)
   }
-
-  const message = optimistic
-    ? await merge(base, pulls, updateComment, updateLabel)
-    : await mergeAll(pulls, updateComment, updateLabel)
+  await exec(`git checkout -B ${branch}`)
 
   // only push to defined branch if there are changes
+  await exec('git fetch')
   if (await hasDiff('HEAD', `origin/${branch}`)) {
-    await exec('git push -f')
+    // ignore any errors
+    await exec('git push -f', undefined, {
+      ignoreReturnCode: true
+    })
   }
-
-  info(message)
 }
 
 const hasDiff = async (a: string, b: string): Promise<boolean> => {
-  let ret = false
-  await exec(`git diff ${a}..${b}`, undefined, {
-    listeners: {
-      stdout: () => {
-        ret = true
-      }
-    }
-  })
-
-  return ret
+  return (
```

Comment:
Since we produce reproducible commit hashes now, just compare them to see if the dev state changed